lib/internal/repl.js

```diff
@@ -120,15 +120,33 @@ function setupHistory(repl, historyPath, oldHistoryPath, ready) {
 
     if (data) {
       repl.history = data.split(/[\n\r]+/, repl.historySize);
-    } else if (oldHistoryPath) {
+    } else if (oldHistoryPath === historyPath) {
+      // If pre-v3.0, the user had set NODE_REPL_HISTORY_FILE to
+      // ~/.node_repl_history. Warn the user about it and proceed.
+      repl._writeToOutput(
+        '\nThe old repl history file has the same name and location as ' +
+        `the new one i.e., ${historyPath} and is empty.\nUsing it as is.\n`);
+      repl._refreshLine();
+
+      // Set repl history to empty array as repl file had no data i.e.,
+      // there was no history.
+      repl.history = [];
```

Comment:
Shouldn't be necessary: https://github.com/nodejs/node/blob/master/lib/readline.js#L144

(`repl` inherits from `readline`)